index.js

```diff
@@ -18,13 +20,13 @@
 
 function paginate(query, options, callback) {
   query = query || {};
-  options = Object.assign({}, paginate.options, options);
-  let select = options.select;
+  options = options || {};
+  let select = options.select || null;
   let sort = options.sort;
   let populate = options.populate;
   let lean = options.lean || false;
-  let leanWithId = options.leanWithId ? options.leanWithId : true;
-  let limit = options.limit ? options.limit : 10;
+  let leanWithId = options.leanWithId || false;
```

Comment:
By default mongoose add `id` field to every document. So if you use it and need `lean` you can get `id` without any special actions